core/src/main/java/org/apache/iceberg/io/DataWriter.java

```diff
@@ -57,6 +56,12 @@ public DataWriter(FileAppender<T> appender, FileFormat format, String location,
     this.sortOrder = sortOrder;
   }
 
+  @Override
+  public void write(T row) {
+    appender.add(row);
+  }
+
+  @Deprecated
```

Comment:
When adding `@Deprecated` annotations, we should always include a comment that states when it will be removed. In this case, `0.14.0` or `1.0.0`.

Reply:
Added.